x/evm/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,487 @@
+				req = &types.QueryBalanceRequest{
+					Address: ethcmn.Address{}.String(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				suite.app.BankKeeper.SetBalance(suite.ctx, suite.address.Bytes(), ethermint.NewPhotonCoinInt64(100))
+				expBalance = "100"
+				req = &types.QueryBalanceRequest{
+					Address: suite.address.String(),
+				}
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+			res, err := suite.queryClient.Balance(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+
+				suite.Require().Equal(expBalance, res.Balance)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestQueryStorage() {
+	var (
+		req      *types.QueryStorageRequest
+		expValue string
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{"zero address",
+			func() {
+				req = &types.QueryStorageRequest{
+					Address: ethcmn.Address{}.String(),
+				}
+			},
+			false,
+		},
+		{"empty hash",
+			func() {
+				req = &types.QueryStorageRequest{
+					Address: suite.address.String(),
+					Key:     ethcmn.Hash{}.String(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				key := ethcmn.BytesToHash([]byte("key"))
+				value := ethcmn.BytesToHash([]byte("value"))
+				expValue = value.String()
+				suite.app.EvmKeeper.SetState(suite.ctx, suite.address, key, value)
+				req = &types.QueryStorageRequest{
+					Address: suite.address.String(),
+					Key:     key.String(),
+				}
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+			res, err := suite.queryClient.Storage(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+
+				suite.Require().Equal(expValue, res.Value)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestQueryCode() {
+	var (
+		req     *types.QueryCodeRequest
+		expCode []byte
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{"zero address",
+			func() {
+				req = &types.QueryCodeRequest{
+					Address: ethcmn.Address{}.String(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				expCode = []byte("code")
+				suite.app.EvmKeeper.SetCode(suite.ctx, suite.address, expCode)
+
+				req = &types.QueryCodeRequest{
+					Address: suite.address.String(),
+				}
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+			res, err := suite.queryClient.Code(ctx, req)
```

Comment:
the comment of this cmd looks like its a copypasta of `Storage(...)` in `x/evm/types/query.pb.proto`